storage/test/sparse_merkle/tree_test.cpp

```diff
@@ -31,19 +36,130 @@ void db_put(const shared_ptr<TestDB>& db, const Tree::UpdateBatch& batch) {
   }
 }
 
-template <class T>
-void assert_version(uint64_t version, const T& keys) {
+template <typename T>
+::testing::AssertionResult keyVersionMatches(uint64_t version, const T& keys) {
```

Comment:
I like the idea of using `AssertionResult`!

Reply:
Thanks! I was getting test failures that always showed the failure coming from the called function that used `ASSERT_XXX` macros and so I couldn't actually tell where in the caller the test was failing, since those called functions were called multiple times in a single test. This lets us actually know which part of the test is failing. Good on gtest for providing a solution.

Reply:
I am going to use that in the future.